pkg/csr/csr.go

```diff
@@ -0,0 +1,260 @@
+		Message:        "This CSR was approved by the WMCO CSR Approver",
+		LastUpdateTime: meta.Now(),
+		Reason:         "WMCOApproved",
+	})
+
+	if _, err := c.k8sClientSet.CertificatesV1().CertificateSigningRequests().UpdateApproval(context.Background(),
+		c.Name, c.CertificateSigningRequest, meta.UpdateOptions{}); err != nil {
+		return errors.Wrapf(err, "could not update conditions for CSR approval %s", c.Name)
+	}
+	c.log.Info("CSR approved by WMCO CSR Approver", "CSR", c.Name)
+	return nil
+}
+
+// validateCSRContents returns true if the CSR request contents are valid
+func (c *Approver) validateCSRContents() (bool, error) {
+	parsedCSR, err := ParseCSR(c.Spec.Request)
+	if err != nil {
+		return false, errors.Wrapf(err, "error parsing CSR %s", c.Name)
+	}
+
+	nodeName := strings.TrimPrefix(parsedCSR.Subject.CommonName, NodeUserNamePrefix)
+	if nodeName == "" {
+		return false, errors.Errorf("CSR subject name does not contain the required node user prefix %s", c.Name)
+	}
+
+	// lookup the node name against the instance hostnames
+	if valid := c.validateNodeName(nodeName); !valid {
+		// CSR is not from a BYOH Windows instance, don't deny it or return error
+		// since it might be from a linux node.
+		return false, nil
+	}
+	// Kubelet on a node needs two certificates for its normal operation:
+	// Client certificate for securely communicating with the Kubernetes API server
+	// Server certificate for use by Kubernetes API server to talk back to kubelet
+	// Both types are validated based on their contents
+	if c.isNodeClientCert(parsedCSR) {
+		// Node client bootstrapper CSR is received before the instance becomes a node
+		// hence we should not proceed if a corresponding node already exists
+		_, err := c.k8sClientSet.CoreV1().Nodes().Get(context.TODO(), nodeName, meta.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, errors.Wrapf(err, "unable to get node %s", nodeName)
+		} else if err == nil {
+			return false, errors.Wrapf(err, "%s node already exists, cannot validate CSR", nodeName)
+		}
+	} else {
+		if validated, err := c.validateKubeletServingCSR(parsedCSR); !validated {
+			return false, errors.Wrapf(err, "unable to validate kubelet serving CSR %s", c.Name)
+		}
+	}
+	return true, nil
+}
+
+// validateNodeName returns true if the node name passed here matches
+// either the actual host name of the VM'S or the reverse lookup of
+// the instance addresses present in the configMap
+func (c *Approver) validateNodeName(nodeName string) bool {
+	if len(c.hostNames) == 0 {
+		c.log.Info("no host names found to validate the CSR")
+		return false
+	}
+	// check if the node name matches host names of the instances
+	for _, hostName := range c.hostNames {
+		if strings.Contains(hostName, nodeName) {
+			return true
+		}
+	}
+	// check if the node name matches the lookup of address in configMap
+	for _, address := range c.instanceAddresses {
+		return c.areAddressesEqual(address, nodeName)
```

Comment:
By returning here won't you be checking with just the first entry in `c.instanceAddresses`?